### PR TITLE
[ADD] feat(file-loader) File loader class for configuration files

### PR DESCRIPTION
**Pull request by Pourbaix**
Added a new file loader class that allows to load a file and read it. It can also be used to determine if the file has change since the last time it was red. By using this class to load files content, it allow us to modify config files without restarting the application.

### CODE REVIEW

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/core/ConfigurationFileLoader.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return this.isError? Files.readAllBytes(Path.of(this.absoluteFilePath)) : null;
+    }
+
+    /**
+     * Public method which indicate if the file has been through modification since the last read.
+     * 
+     * @return - TRUE if it has been modified since the list read.
+     *         - FALSE if it has not been modified since the last read.
+     */
+    public boolean isFileNewer(){
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/core/ConfigurationFileLoader.java:69]: `isFileNewer()` --> `modified()` ?

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/core/ConfigurationFileLoader.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     */
+    private void loadFile(){
+        File configurationFile = new File(this.absoluteFilePath);
+        if (configurationFile.exists() && configurationFile.canRead()){
+            this.lastModified = configurationFile.lastModified();
+            this.isError = false;
+        } else {
+            this.isError = true;
+        }
+    }
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/core/ConfigurationFileLoader.java:37]: Don't think it's a smart idea to keep `isError` flag ; just throws a `IOException` is files doesn't exists or can't be read.

**Review by zannkukai — CHANGES_REQUESTED** [review]
I don't understand where is the benefit of this class. You never really "reload" the file if it has been updated since last access. 
When you get data, you use File.readString|readAsByte that open the file, read the content, ... each time.

It think you need to use "factory pattern" in this case.
The factory loader class return instance of "ConfigurationFile" (that should be singleton classes themself). On "ConfigurationFile", you could cached the file content at first call of "read" functions. At second call, if file is updated, reload the content and update the cache ; otherwise return cache directly.

We could discuss about this ; It will be easier

**Comment by Pourbaix** [comment]
Bullshit commit :wastebasket: will close and redo new PR after.

**Pull request closed without merging**